x-pack/platform/packages/shared/ai-assistant/ai-assistant-connector-selector-action/connector_selector.tsx

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo, useState } from 'react';
+import type { EuiSelectableOption } from '@elastic/eui';
+import {
+  EuiBadge,
+  EuiButton,
+  EuiButtonIcon,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIcon,
+  EuiPanel,
+  EuiPopoverFooter,
+  EuiSelectable,
+} from '@elastic/eui';
+import type { EuiSelectableLIOption } from '@elastic/eui/src/components/selectable/selectable_option';
+import { translations as i8n } from './connector_selector.translations';
+
+/** Props for the `ConnectorSelector` */
+export interface ConnectorSelectorProps<T extends { value: string } = { value: string }> {
+  /** Pre-configured connectors to display in the selector */
+  preConfiguredConnectors: (EuiSelectableLIOption<T> & { key?: undefined; checked?: undefined })[];
+  /** Custom connectors to display in the selector */
+  customConnectors: (EuiSelectableLIOption<T> & { key?: undefined; checked?: undefined })[];
+  /** Default connector id if a default connector has been configured. */
+  defaultConnectorId?: string;
+  /** Optional test subject for the component. */
+  ['data-test-subj']?: string;
+  /** Controlled selected connector value. */
+  value?: string;
+  /** Uncontrolled initial selected connector value. */
+  defaultValue?: string;
+  /** Callback that provides the selected connector value (string). */
+  onValueChange?: (value: string, option: EuiSelectableOption<T>) => void;
+  /** Label for the "Add Connector" button. Default is "Add Connector" */
+  addConnectorLabel?: string;
+  /** Aria label for the "Add Connector" button. Default is "Add Connector" */
+  addConnectorAriaLabel?: string;
```

Comment:
nit: is there value in all of these labels being exposed?  They have reasonable defaults, so I would propose only exposing those that _must_ be customizable, and exposing as they become necessary.

In a shared component, the more you expose, the more you have to support/keep around.  Go for the absolute minimum, for now, for all of these props.

It's also less for a consumer to think about.